packages/playwright-core/src/server/injected/webSocketMock.ts

```diff
@@ -332,6 +332,13 @@ export function inject(globalThis: GlobalThis) {
       if (this.readyState !== WebSocketMock.CONNECTING)
         return;
       this.readyState = WebSocketMock.OPEN;
+      this.protocol = (() => {
```

Comment:
- I don't quite like IIFE when we don't really need it. Can we do if/else chain instead?
- What about `WebSocket.extensions`? Perhaps we should do the same.

Reply:
1. Done.
2. `WebSocket.extensions` are getting handled by the browser, hence it looks simpler there. should I add a test for it? It would require to enable deflate conditionally in the WS server in `tests/config/testserver/index.ts:81`. Maybe create a WS server inline?